### PR TITLE
Fix empty collection saved in ds table

### PR DESCRIPTION
**Pull request by StefanKlt**

### CODE REVIEW

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/distributions/dstable/DistributionSetTable.java [hunk 1]
```diff
@@ -230,26 +230,53 @@ protected boolean isDropValid(final DragAndDropEvent dragEvent) {
...
-            openConfirmationWindowForAssignment(distributionSetIdName.getName(),
-                    softwareModules.toArray(new SoftwareModuleIdName[softwareModules.size()]));
-        });
         if (!distributionSet.isPresent()) {
             getNotification().displayWarning(getI18n().getMessage("distributionset.not.exists"));
+            return;
         }
+        distributionSet.ifPresent(ds -> {
+            selectDroppedEntities(ds.getId());
+            final HashSet<SoftwareModule> softwareModules = getAssignableSoftwareModules(softwareModulesIdList, ds);
```
Comment (StefanKlt) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/distributions/dstable/DistributionSetTable.java]: use Set

**Comment by schabdo** [comment]
@hawkbit-bot verify please

**Comment by stefbehl** [comment]
@hawkbit-bot verify please

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/distributions/dstable/DistributionSetTable.java [hunk 2]
```diff
@@ -230,26 +231,55 @@ protected boolean isDropValid(final DragAndDropEvent dragEvent) {
...
             getNotification().displayWarning(getI18n().getMessage("distributionset.not.exists"));
+            return;
+        }
+        distributionSet.ifPresent(ds -> {
+            selectDroppedEntities(ds.getId());
+            final Set<SoftwareModule> softwareModules = getAssignableSoftwareModules(softwareModulesIdList, ds);
+            if (softwareModules.isEmpty()) {
+                return;
+            }
+            assignSoftwareModulesToDs(softwareModules, ds);
```
Comment (stefbehl) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/distributions/dstable/DistributionSetTable.java]: You could reduce this to
            openConfirmationWindowForAssignment(ds.getName(),
                    softwareModules.stream().map(SoftwareModuleIdName::new)
                            .toArray(size -> new SoftwareModuleIdName[size]));

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/distributions/dstable/DistributionSetTable.java [hunk 3]
```diff
@@ -230,26 +231,55 @@ protected boolean isDropValid(final DragAndDropEvent dragEvent) {
...
+        return assignableModules;
     }
 
-    private HashMap<Long, HashSet<SoftwareModuleIdName>> createAssignmentMap(
-            final DistributionSetIdName distributionSetIdName) {
+    private void assignSoftwareModulesToDs(final Set<SoftwareModule> softwareModules,
+            final DistributionSet distributionSet) {
+        addSoftwareModulesToConsolidatedDsAssignmentMap(distributionSet, softwareModules);
+
+        final HashSet<SoftwareModuleIdName> softwareModulesThatNeedToBeAssigned = new HashSet<>();
```
Comment (stefbehl) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/distributions/dstable/DistributionSetTable.java]: final Set<...>

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/distributions/dstable/DistributionSetTable.java [hunk 4]
```diff
@@ -367,44 +380,39 @@ private void resetState() {
...
             /* Invalid type of the software module */
             getNotification().displayValidationError(getI18n().getMessage("message.software.dist.type.notallowed",
-                    HawkbitCommonUtil.concatStrings(":", sm.getName(), sm.getVersion()),
-                    HawkbitCommonUtil.concatStrings(":", ds.getName(), ds.getVersion()), sm.getType().getName()));
+                    smNameAndVersion, dsNameAndVersion, sm.getType().getName()));
+            return false;
+        }
+        if (distributionSetManagement.isInUse(ds.getId())) {
+            getNotification()
+                    .displayValidationError(getI18n().getMessage("message.error.notification.ds.target.assigned",
```
Comment (stefbehl) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/distributions/dstable/DistributionSetTable.java]: remove unnecessary line break

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/entity/SoftwareModuleIdName.java [hunk 5]
```diff
@@ -22,16 +22,30 @@
 
     private final Long id;
     private final String name;
+    private final String version;
 
     /**
      * @param id
-     *            if the {@link SoftwareModule}
+     *            if the {@link SoftwareModule#getId()}
```
Comment (stefbehl) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/entity/SoftwareModuleIdName.java]: "of the {@link SoftwareModule}" ... same for next two params

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/distributions/dstable/DistributionSetTable.java (hunk 2 above)
Comment (StefanKlt) [reply hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/distributions/dstable/DistributionSetTable.java]: fixed

**Pull request merged**